src/main/java/com/spotifyapi/service/impl/UserServiceImpl.java

```diff
@@ -1,33 +1,40 @@
 package com.spotifyapi.service.impl;
 
 
-import com.spotifyapi.service.UserService;
+import com.spotifyapi.model.User;
+import com.spotifyapi.service.UserRepository;
 import lombok.AllArgsConstructor;
 import org.apache.hc.core5.http.ParseException;
 import org.springframework.stereotype.Service;
 import se.michaelthelin.spotify.SpotifyApi;
 import se.michaelthelin.spotify.exceptions.SpotifyWebApiException;
 
 import java.io.IOException;
+import java.util.List;
 
 @Service
 @AllArgsConstructor
-public class UserServiceImpl implements UserService {
+public class UserServiceImpl {
 
     private final SpotifyApi spotifyApi;
+    private final UserRepository userRepository;
 
-    @Override
-    public String getUserData() {
+    public void saveUserData(List<String> tokens) {
         try {
+            User newUser = new User();
+
             var userProfile = spotifyApi.getCurrentUsersProfile().build().execute();
-            return "You are get your data successfully! \n"
-                    + "User name: " + userProfile.getDisplayName() + " \n"
-                    + "Email: " + userProfile.getEmail() + " \n"
-                    + "Id: " + userProfile.getId() + " \n";
+
+            newUser.setUsername(userProfile.getDisplayName());
+            newUser.setEmail(userProfile.getEmail());
+            newUser.setSpotifyUserId(userProfile.getId());
+            newUser.setAccessToken(tokens.get(0));
```

Comment:
such approach with list can lead to errors

Reply:
Created class TokensDTO instead of List